app/api/views.py

```diff
@@ -13,5 +13,13 @@ def process_coords():
     Returns: This returns a post request of create_segments().
     """
 
-    data = json.loads(request.form.get("coords")).get("b")
-    return "{}".format(create_segments(data))
+    data = json.loads(request.form.get("coords"))
+    hsegs = []
+
+    for polygon in data:
+    	segment = polygon.get("b")
+    	hsegs.append(segment)
+
+    session["polygons"] = hsegs
+    
+    return "{}".format(session)
```

Comment:
* Region isn't being created for each polygon
* You're returning the session

I'm thinking there should be an entry in `session` for the regions that are generated from `create_segments` like how `polygon` entry is. `create_segments` should be refactored to be called `create_region`.